featurebyte/routes/observation_table/controller.py

```diff
@@ -110,22 +113,38 @@ async def upload_observation_table(
             if the observation set file format is not supported
         """
         assert observation_set_file.filename is not None
-        file_format = UploadFileFormat.CSV
-        if observation_set_file.filename.lower().endswith(".csv"):
-            assert observation_set_file.content_type == "text/csv"
-            observation_set_dataframe = pd.read_csv(observation_set_file.file)
+
+        def try_read_as_dataframe(
+            read_func: Callable[[str], pd.DataFrame], *args: Any
+        ) -> pd.DataFrame:
+            try:
+                return read_func(*args)
+            except Exception as exc:
+                raise UnsupportedObservationTableUploadFileFormat(
+                    "Content of uploaded file is not valid"
+                ) from exc
+
+        filename = observation_set_file.filename.lower()
+        if not filename.endswith(".csv") and not filename.endswith(".parquet"):
+            raise UnsupportedObservationTableUploadFileFormat(
+                "Only csv and parquet file formats are supported for observation set upload"
+            )
+
+        if filename.endswith(".csv"):
+            file_format = UploadFileFormat.CSV
+            observation_set_dataframe = try_read_as_dataframe(
+                pd.read_csv, observation_set_file.file
+            )
             # Convert point_in_time column to datetime
             observation_set_dataframe[SpecialColumnName.POINT_IN_TIME] = pd.to_datetime(
                 observation_set_dataframe[SpecialColumnName.POINT_IN_TIME]
             )
-        elif observation_set_file.filename.lower().endswith(".parquet"):
-            file_format = UploadFileFormat.PARQUET
-            assert observation_set_file.content_type == "application/octet-stream"
```

Comment:
The `Content-Type` header is not set by actual API clients outside of tests, so this assertion failed.